### PR TITLE
Switch to gunicorn, optimise Dockerfile and split dev dependencies

### PR DESCRIPTION
**Pull request by Sytten**
Changes:
- Switch to python slim (alpine isn't really recommended)
- Switch to gunicorn with a basic configuration (4 processes, 4 threads per process)
- Split dev dependencies from prod dependencies
- Switch sh to bash

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
- apk add --no-cache --virtual .build-deps gcc musl-dev postgresql-dev && \
- pip install -r requirements.txt --no-cache-dir && \
- apk --purge del .build-deps
-
-COPY . /JDISCTF
-
-RUN ["mv", ".env.production", ".env"]
+COPY jdisctf.py .
+COPY config.py .
+COPY config.py .
```
Comment (Res260) [inline backend/Dockerfile]: euh

File: backend/Dockerfile [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
 
-RUN mkdir /JDISCTF
-WORKDIR /JDISCTF
+WORKDIR /flaggr
 
-COPY requirements.txt /JDISCTF
+COPY requirements.txt .
+RUN pip install gunicorn==19.10.0 --no-cache-dir
+RUN pip install -r requirements.txt --no-cache-dir && \
+ find /usr/local \
```
Comment (Blond11516) [inline backend/Dockerfile]: What does this do?

Comment (Sytten) [reply backend/Dockerfile]: It removes all compiled python stuff (.pyc) and tests from the dependencies

**Comment by Res260** [comment]
The admin section doesnt load:
image

**Comment by Res260** [comment]
I also keep getting those:

image

**Comment by Sytten** [comment]
Probably a small bug in loading the env

**Pull request merged**